aws/resource_aws_dx_hosted_private_virtual_interface_accepter.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDxHostedPrivateVirtualInterfaceAccepter() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDxHostedPrivateVirtualInterfaceAccepterCreate,
+		Read:   resourceAwsDxHostedPrivateVirtualInterfaceAccepterRead,
+		Update: resourceAwsDxHostedPrivateVirtualInterfaceAccepterUpdate,
+		Delete: resourceAwsDxHostedPrivateVirtualInterfaceAccepterDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceAwsDxHostedPrivateVirtualInterfaceAccepterImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"virtual_interface_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vpn_gateway_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"dx_gateway_id"},
+			},
+			"dx_gateway_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"vpn_gateway_id"},
+			},
+			"tags": tagsSchema(),
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+	}
+}
+
+func resourceAwsDxHostedPrivateVirtualInterfaceAccepterCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	vgwIdRaw, vgwOk := d.GetOk("vpn_gateway_id")
+	dxgwIdRaw, dxgwOk := d.GetOk("dx_gateway_id")
+	if vgwOk == dxgwOk {
+		return fmt.Errorf(
+			"One of ['vpn_gateway_id', 'dx_gateway_id'] must be set to create a Direct Connect private virtual interface accepter")
+	}
+
+	vifId := d.Get("virtual_interface_id").(string)
+	req := &directconnect.ConfirmPrivateVirtualInterfaceInput{
+		VirtualInterfaceId: aws.String(vifId),
+	}
+	if vgwOk && vgwIdRaw.(string) != "" {
+		req.VirtualGatewayId = aws.String(vgwIdRaw.(string))
+	}
+	if dxgwOk && dxgwIdRaw.(string) != "" {
+		req.DirectConnectGatewayId = aws.String(dxgwIdRaw.(string))
+	}
+
+	log.Printf("[DEBUG] Accepting Direct Connect hosted private virtual interface: %#v", req)
+	_, err := conn.ConfirmPrivateVirtualInterface(req)
+	if err != nil {
+		return fmt.Errorf("Error accepting Direct Connect hosted private virtual interface: %s", err.Error())
+	}
+
+	d.SetId(vifId)
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Service:   "directconnect",
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("dxvif/%s", d.Id()),
+	}.String()
+	d.Set("arn", arn)
+
+	if err := dxHostedPrivateVirtualInterfaceAccepterWaitUntilAvailable(d, conn); err != nil {
+		return err
+	}
+
+	return resourceAwsDxHostedPrivateVirtualInterfaceAccepterUpdate(d, meta)
+}
+
+func resourceAwsDxHostedPrivateVirtualInterfaceAccepterRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
The read function doesn't seem to be checking the VIF state to trigger resource recreation (e.g. ensure the VIF state is `available`/`down`) if its for some reason disabled.

Reply:
OK, I'll address this.

Reply:
Yes, I think once we no longer delete on the accepting side, if the resource is in an "acceptable" state during `Read` we should clear the `Id` and trigger a re-acceptance.